.github/workflows/style.yml

```diff
@@ -9,18 +9,18 @@ on:
     - main
     - release**
 jobs:
-  black:
-    name: black
+  ruff:
+    name: ruff
     runs-on: ubuntu-latest
     steps:
     - name: Clone repo
-      uses: actions/checkout@v4.1.1
+      uses: actions/checkout@v4.1.2
     - name: Set up python
-      uses: actions/setup-python@v5.0.0
+      uses: actions/setup-python@v5.1.0
       with:
-        python-version: '3.11'
+        python-version: '3.12'
     - name: Cache dependencies
-      uses: actions/cache@v4.0.0
+      uses: actions/cache@v4.0.2
```

Comment:
Most of these version bumps should be saved for another PR, probably by dependabot